### PR TITLE
jsx-alignment rule implementation

### LINKED ISSUE #7: New Rule: jsx-alignment
Enforce this alignment when tags span multiple lines:

```ts
<input
    type="search"
    placeholder="Search..."
/>
```

### PR DESCRIPTION
**Pull request by JKillian**
Closes #7

### CODE REVIEW

**Comment by JKillian** [comment]
I think this is a solid starting state for this rule, figured it'll be good to go ahead and release it to make it easier to use on actual codebases.

File: README.md [hunk 1]
```diff
@@ -26,6 +26,26 @@ Sample configuration where `tslint.json` lives adjacent to your `node_modules` f
 
 ### Rules
 
+- `jsx-alignment`
+  - Enforces a consistent style for multiline JSX elements which promotes ease of editing via line-wise manipulations
+  as well as maintainabilty via small diffs when changes are made.
+  ```
```
Comment (adidahiya) [inline README.md]: add `ts` at the end of this line to get syntax highlighting

File: src/rules/jsxAlignmentRule.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+}
+
+const leadingWhitespace = /[ \t]/;
+
+class JsxAlignmentWalker extends Lint.RuleWalker {
+    protected visitJsxElement(node: ts.JsxElement) {
+        if (this.isMultiline(node)) {
+            const startLocation = this.getLineAndCharacter(node);
+            const closeLocation = this.getSourceFile().getLineAndCharacterOfPosition(
+                node.openingElement.getEnd() - ">".length
```
Comment (adidahiya) [inline src/rules/jsxAlignmentRule.ts]: at a glance this looks like an arrow function :D

File: src/rules/jsxAlignmentRule.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    public static TAG_CLOSE_FAILURE = "Tag closing must be on its own line and aligned with opening of tag";
+    public static CLOSING_TAG_FAILURE = "Closing tag must be on its own line and aligned with opening tag";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const walker = new JsxAlignmentWalker(sourceFile, this.getOptions());
+        return this.applyWithWalker(walker);
+    }
+}
+
+const leadingWhitespace = /[ \t]/;
```
Comment (adidahiya) [inline src/rules/jsxAlignmentRule.ts]: minor: I like suffixing names of regex values with "Regex"

File: src/rules/jsxAlignmentRule.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+    private getLineAndCharacter(node: ts.Node) {
+        const sourceFile = this.getSourceFile();
+        return sourceFile.getLineAndCharacterOfPosition(node.getStart(sourceFile));
+    }
+
+    private getCharacter = (node: ts.Node) => this.getLineAndCharacter(node).character;
+    private getLine = (node: ts.Node) => this.getLineAndCharacter(node).line;
+
+    private nodeFailure(node: ts.Node, message: string) {
```
Comment (adidahiya) [inline src/rules/jsxAlignmentRule.ts]: minor: can we call this `reportFailure` for consistency with some other rules?

**Comment by adidahiya** [comment]
Overall looks good, just left some minor comments.

**Pull request merged**